tests/integration/test_database.py

```diff
@@ -87,6 +95,97 @@ async def test_build_and_deploy(ops_test: OpsTest):
 @pytest.mark.order(2)
 @pytest.mark.abort_on_fail
 @pytest.mark.database_tests
+async def test_password_rotation(ops_test: OpsTest):
+    """Rotate password and confirm changes."""
+    random_unit = ops_test.model.applications[DATABASE_APP_NAME].units[-1]
+
+    old_credentials = await fetch_credentials(random_unit, CLUSTER_ADMIN_USERNAME)
+
+    # get primary unit first, need that to invoke set-password action
+    primary_unit = await get_primary_unit(ops_test, random_unit, DATABASE_APP_NAME)
+    primary_unit_address = await primary_unit.get_public_address()
+    logger.debug(
+        "Test succeeded Primary unit detected before password rotation is %s", primary_unit_address
+    )
+
+    new_password = generate_random_password(PASSWORD_LENGTH)
+
+    await rotate_credentials(
+        unit=primary_unit, username=CLUSTER_ADMIN_USERNAME, password=new_password
+    )
+
+    updated_credentials = await fetch_credentials(random_unit, CLUSTER_ADMIN_USERNAME)
+    assert updated_credentials["password"] != old_credentials["password"]
+    assert updated_credentials["password"] == new_password
+
+    # verify that the new password actually works
```

Comment:
nit: please add a comment indicating that the cluster_admin credentials are used to get the primary unit (when calling get_cluster_status)